AsyncUsageAnalyzers/AsyncUsageAnalyzers.CodeFixes/Usage/DontUseThreadSleepCodeUniversalCodeFixProvider.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+namespace AsyncUsageAnalyzers.Usage
+{
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Helpers;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Text;
+
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(DontUseThreadSleepCodeUniversalCodeFixProvider))]
+    [Shared]
+    internal class DontUseThreadSleepCodeUniversalCodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+                ImmutableArray.Create(DontUseThreadSleepAnalyzer.DiagnosticId, DontUseThreadSleepInAsyncCodeAnalyzer.DiagnosticId);
+
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return CustomFixAllProviders.BatchFixer;
+        }
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                if (diagnostic.Id == DontUseThreadSleepInAsyncCodeAnalyzer.DiagnosticId)
+                {
+                    RegisterCodeFixForDiagnostic(context, diagnostic);
+                }
+                else if (diagnostic.Id == DontUseThreadSleepAnalyzer.DiagnosticId)
+                {
+                    var document = context.Document;
+
+                    var root = await document.GetSyntaxRootAsync().ConfigureAwait(false);
+                    var invocationExpression = root.FindNode(TextSpan.FromBounds(diagnostic.Location.SourceSpan.Start, diagnostic.Location.SourceSpan.End), getInnermostNodeForTie: true) as InvocationExpressionSyntax;
+
+                    if (invocationExpression == null)
+                    {
+                        return;
+                    }
+
+                    if (invocationExpression.IsInsideAsyncCode())
+                    {
+                        RegisterCodeFixForDiagnostic(context, diagnostic);
+                    }
+                }
+            }
+        }
+
+        private static void RegisterCodeFixForDiagnostic(CodeFixContext context, Diagnostic diagnostic)
+        {
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    "Use `await Task.Delay(...)` or `await Task.Yield()`",
```

Comment:
No idea if it is worth it, just looked a bit strange with a roulette type code fix :)